### PR TITLE
Feature/traits

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

1st commit is to factorize `post_clone`, which were all the same but relations classes list.
2nd and 3rd and 4th commits just make use of the new namespace.

### CODE REVIEW

File: inc/appliance.class.php [hunk 1]
```diff
@@ -40,6 +40,16 @@ class Appliance extends CommonDBTM {
    static $rightname     = "appliance";
    protected $usenotepad = true;
 
+   /** RELATIONS */
+   public function getCloneRelations() :array {
```
Comment (cedric-anne) [inline inc/appliance.class.php:44]: Missing `use Glpi\Features\Clonable;` ?

File: inc/features/clonable.class.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+   die("Sorry. You can't access this file directly");
+}
+
+use CommonDBConnexity;
+use Toolbox;
+
+/**
+ * Clonable objects
+ **/
+trait clonable {
```
Comment (cedric-anne) [inline inc/features/clonable.class.php:45]:
```suggestion
trait Clonable {
```

File: inc/features/clonable.class.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   /**
+    * Get relations class to clone along with current eleemnt
+    *
+    * @return CommonDBTM::class[]
+    */
+   abstract public function getCloneRelations() :array;
```
Comment (cedric-anne) [inline inc/features/clonable.class.php:51]: Should the method be declared as protected ?

Comment (trasher) [reply inc/features/clonable.class.php:51]: No, even if it is not designed to be usefull outside the feature; it does not seems a problem to be public.

**Comment by trasher** [comment]
For 1- `Glpi\Items\Traits`, well, no, this brings no semantic at all.
And for 2-... well, I'm not sure I do care about that :D

**Comment by trasher** [comment]
> For 1- `Glpi\Items\Traits`, well, no, this brings no semantic at all.
Also, this is not intended to provide only traits, this is currently a fact, but this is not the goal at all.

**Comment by trasher** [comment]
"all commondbtm items seems to be clonable" => not sure about that (event if it is not important).

As for the name, well, I do not have any other proposal for the moment

**Pull request merged**